### PR TITLE
feat: Added purchase receipt field in stock entry doctype

### PR DESCRIPTION
**Pull request by meshramsaroj**
**Task Link:** https://app.asana.com/0/0/1199600812134038/f

**Description:** Added purchase receipt field in stock entry doctype so that we can fetch the stock entry's item table data and display it on purchase invoice print format.

**Screenshot:** 
image

### CODE REVIEW

File: erpnext/stock/doctype/stock_entry/stock_entry.json [hunk 1]
```diff
@@ -638,12 +639,19 @@
    "label": "Harvest",
    "options": "Harvest",
    "read_only": 1
+  },
+  {
+   "fetch_from": "work_order.purchase_receipt",
+   "fieldname": "purchase_receipt",
+   "fieldtype": "Data",
```
Comment (vishdha) [inline erpnext/stock/doctype/stock_entry/stock_entry.json]: field type should be read only

**Review by vishdha — CHANGES_REQUESTED** [review]

File: erpnext/stock/doctype/stock_entry/stock_entry.json (hunk 1 above)
Comment (meshramsaroj) [reply erpnext/stock/doctype/stock_entry/stock_entry.json]: filed type is read-only see line number 647 and 648

File: erpnext/stock/doctype/stock_entry/stock_entry.json [hunk 2]
```diff
@@ -638,12 +639,19 @@
    "label": "Harvest",
    "options": "Harvest",
    "read_only": 1
+  },
+  {
+   "fetch_from": "work_order.purchase_receipt",
+   "fieldname": "purchase_receipt",
+   "fieldtype": "Read Only",
+   "label": "Purchase Receipt",
+   "read_only": 1
```
Comment (vishdha) [inline erpnext/stock/doctype/stock_entry/stock_entry.json]: remove this check 
```suggestion
  
```

**Review by vishdha — CHANGES_REQUESTED** [review]

File: erpnext/stock/doctype/stock_entry/stock_entry.json [hunk 3]
```diff
@@ -638,12 +639,19 @@
    "label": "Harvest",
    "options": "Harvest",
    "read_only": 1
+  },
+  {
+   "fetch_from": "work_order.purchase_receipt",
+   "fieldname": "purchase_receipt",
+   "fieldtype": "Read Only",
+   "label": "Purchase Receipt",
+
```
Comment (vishdha) [inline erpnext/stock/doctype/stock_entry/stock_entry.json]:
```suggestion
```

**Review by vishdha — CHANGES_REQUESTED** [review]

**Pull request merged**